File: javaproject/app/src/main/java/com/marcosvbras/robomarket/app/BaseActivity.java
package com.marcosvbras.robomarket.app;

import android.content.Intent;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.marcosvbras.robomarket.R;
import com.marcosvbras.robomarket.interfaces.BaseActivityCallback;

public class BaseActivity extends AppCompatActivity implements BaseActivityCallback {

    private AlertDialog.Builder alertDialog;

    @Override
    public void showDialogMessage(String message) {
        showAlertDialog(message);
    }

    @Override
    public void showDialogMessage(int message) {
        showAlertDialog(getString(message));
    }

    private void showAlertDialog(String message) {
        if (alertDialog == null)
            alertDialog = new AlertDialog.Builder(this);

        alertDialog
                .setMessage(message)
                .setPositiveButton(getString(R.string.ok), null)
                .show();
    }

    @Override
    public void showCustomAlertDialog(DialogFragment dialogFragment) {
        FragmentManager fragmentManager = getSupportFragmentManager();
        dialogFragment.show(fragmentManager, dialogFragment.getTag());
        fragmentManager.beginTransaction().commitAllowingStateLoss();
    }

    @Override
    public void openActivity(Class<?> activity, boolean finishCurrentActivity) {
        startActivity(new Intent(this, activity));

        if(finishCurrentActivity)
            finish();
    }

    @Override
    public void openActivity(Class<?> activity, Bundle bundle, boolean finishCurrentActivity) {
        Intent intent = new Intent(this, activity);

        if(bundle != null)
            intent.putExtras(bundle);

        startActivity(intent);

        if(finishCurrentActivity)
            finish();
    }

    @Override
    public void openActivityForResult(Class<?> activity, Bundle bundle, int requestCode) {
        Intent intent = new Intent(this, activity);

        if(bundle != null)
            intent.putExtras(bundle);

        startActivityForResult(intent, requestCode);
    }

    @Override
    public void setToolbar(int viewId, boolean displayHomeAsUpEnabled) {
        setSupportActionBar(findViewById(viewId));
        getSupportActionBar().setDisplayHomeAsUpEnabled(displayHomeAsUpEnabled);
    }

    @Override
    public void finishCurrentActivity() {
        finish();
    }

    @Override
    public void setActivityResult(int resultCode) {
        setResult(resultCode);
    }

    @Override
    public void setActivityResult(int resultCode, Intent intent) {
        setResult(resultCode, intent);
    }

    @Override
    public void setActivityResult(int resultCode, Bundle bundle) {
        Intent intent = getIntent();

        if(bundle != null)
            intent.putExtras(bundle);

        setResult(resultCode, intent);
    }
}

File: javaproject/app/src/main/java/com/marcosvbras/robomarket/app/BaseActivity.kt
package com.marcosvbras.robomarket.app

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v4.app.ActivityCompat
import android.support.v4.app.ActivityOptionsCompat
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import com.marcosvbras.robomarket.R
import com.marcosvbras.robomarket.interfaces.BaseActivityCallback

open class BaseActivity : AppCompatActivity(), BaseActivityCallback {

    private var alertDialog: AlertDialog.Builder? = null

    override fun showDialogMessage(message: String) {
        showAlertDialog(message)
    }

    override fun showDialogMessage(message: Int) {
        showAlertDialog(getString(message))
    }

    private fun showAlertDialog(message: String) {
        if (alertDialog == null)
            alertDialog = AlertDialog.Builder(this)

        alertDialog!!
                .setMessage(message)
                .setPositiveButton(getString(R.string.ok), null)
                .show()
    }

    override fun showCustomAlertDialog(dialogFragment: DialogFragment) {
        val fragmentManager = supportFragmentManager
        dialogFragment.show(fragmentManager, dialogFragment.tag)
        fragmentManager.beginTransaction().commitAllowingStateLoss()
    }

    override fun openActivity(activity: Class<*>, bundle: Bundle?, finishCurrentActivity: Boolean) {
        val intent = Intent(this, activity)

        if (bundle != null)
            intent.putExtras(bundle)

        startActivity(intent)

        if (finishCurrentActivity)
            finish()
    }

    override fun openActivityWithAnimation(activity: Class<*>, bundle: Bundle?, finishCurrentActivity: Boolean, enterAnimation: Int, exitAnimation: Int) {
        val intent = Intent(this, activity)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK

        if (bundle != null)
            intent.putExtras(bundle)

        val opts = ActivityOptionsCompat.makeCustomAnimation(this, enterAnimation, exitAnimation)
        ActivityCompat.startActivity(this, intent, opts.toBundle())

        if (finishCurrentActivity)
            finish()
    }

    override fun openActivityForResult(activity: Class<*>, bundle: Bundle?, requestCode: Int) {
        val intent = Intent(this, activity)

        if (bundle != null)
            intent.putExtras(bundle)

        startActivityForResult(intent, requestCode)
    }

    override fun setToolbar(viewId: Int, displayHomeAsUpEnabled: Boolean) {
        setSupportActionBar(findViewById(viewId))
        supportActionBar!!.setDisplayHomeAsUpEnabled(displayHomeAsUpEnabled)
    }

    override fun finishCurrentActivity() {
        finish()
    }

    override fun setActivityResult(resultCode: Int) {
        setResult(resultCode)
    }

    override fun setActivityResult(resultCode: Int, intent: Intent) {
        setResult(resultCode, intent)
    }

    override fun setActivityResult(resultCode: Int, bundle: Bundle) {
        val intent = intent

        if (bundle != null)
            intent.putExtras(bundle)

        setResult(resultCode, intent)
    }

    override fun showSnackBar(message: String, length: Int) {
        Snackbar.make(findViewById(android.R.id.content), message, length).show()
    }

    override fun showSnackBar(message: Int, length: Int) {
        Snackbar.make(findViewById(android.R.id.content), getString(message), length).show()
    }
}